### PR TITLE
107 microservices

### PR DESCRIPTION
**Pull request by FodT**
Some new things happening here to support the new Deals service and the deprecation of the old endpoints

The `strategies` entity now allows for `save_deals()` alongside `save_supplies()`- the syntax here is the same as saving supplies.  to use, call the strategy with `child=deals` (optional), then call strategy.save_deals with `data=[dict of deals]` 

Asking for a `deal` entity now talks to the new media API and pulls the json entity. saving will also post to this endpoint in json format.

entities are now aware of whether they should be posting formdata or json.

### CODE REVIEW

File: terminalone/jsonparser.py [hunk 1]
```diff
@@ -59,7 +59,10 @@ def __init__(self, body):
                 data.get('enabled') is not None:
             self.entities = self._parse_target_dimensions(data)
 
-        elif data.get('permissions') is not None:
+        # FIXME: permissions responses dont have an 'entity_type' field so we're using the
+        # FIXME: absence of this field to identify whether it's a permissions object
+        # FIXME: or just a field named 'permissions'. As always, negative tests are bad. baaaaad.
```
Comment (fsargent) [inline terminalone/jsonparser.py]: Ouch. Is there a way to fix this without an API change?

File: terminalone/t1types.py [hunk 2]
```diff
@@ -49,11 +50,16 @@ def get_value(test_value):
 
 def strpt(dt_string):
```
Comment (fsargent) [inline terminalone/t1types.py]: Please add additional testing here.
image
https://codeclimate.com/github/MediaMath/t1-python/coverage/587550e9323363000109ef8f

Comment (fsargent) [inline terminalone/t1types.py]: Nevermind, I see you did! Great! LGTM.

**Comment by FodT** [comment]
hi @thomas91310, you're right! I need to do a little work to support this. Apologies.

**Comment by FodT** [comment]
@thomas91310 can you check out my branch at https://github.com/FodT/t1-python/tree/107-child-deals ? 
We might need to retire the xml parser soon, since there's a very, very dirty hack there to make this work.

**Pull request merged**